0015-estimator-interface.md

```diff
@@ -0,0 +1,457 @@
+            if circuits is not None:
+                raise ValueError("Cannot mix old and new APIs")
+            circuits = tasks
+            tasks = None
+
+        # Do not mix new/old API
+        if tasks and (circuits or observables or parameter_values):
+            raise ValueError("Cannot mix old and new APIs")
+
+        # Deprecated path
+        if tasks is None:
+            # Verify values in old API
+            if not (circuits and observables):
+                raise ValueError(f"Need both circuits and observables for old API, got circuits={circuits}, observables={observables}")
+            
+            # Coerce into old form
+            if parameter_values is None:
+                parameter_values = itertools.repeat(None)
+            if isinstance(circuits, QuantumCircuit):
+                circuits = [circuits]
+            if isinstance(observables, (BaseOperator, str)):
+                observables = [observables]
+
+            # Coerce old form into `EstimatorTask`s
+            tasks = [EstimatorTask.coerce(task) for task in zip(circuits, observables, parameter_values)]
+            warnings.warn("Deprecated API use")
+        
+        # Coerce into sequence form
+        if isinstance(tasks, EstimatorTasks):
+            tasks = [tasks]
+        
+        # Run using only tasks
+        return self._run(tasks, **run_options)
+```
+
+### Deprecation Phase 2
+
+After the deprecation period, remove extraneous arguments and checks:
+
+```python
+class Estimator(BasePrimitive):
+
+    def run(
+        self, 
+        tasks: Sequence[EstimatorTask] | EstimatorTask, 
+        **run_options
+    ):
+        # Coerce into sequence form
+        if isinstance(tasks, EstimatorTasks):
+            tasks = [tasks]
+        
+        # Run using only tasks
+        return self._run(tasks, **run_options)
+```
+
+## Alternative Approaches <a name="alternative-approaches"></a>
+
+An alternative is to consider letting the `run()` method accept, effectively, only a single `EstimatorTask`:
+
+```python
+Estimator.run(cirucuit, parameter_values_array, observables_array)
+```
+
+This has the advantage of a simpler interface, where multiple tasks could be run 
+by invoking the estimator multiple times. The disadvantages, which we feel are significant enough to forego this simplification, are that:
+
+ 1. For real backends, the user would lose the ability to cause multiple types of circuits to be loaded into the control hardware at one time. For example, if using an estimator to perform randomized benchmarking, each circuit depth would need to be a separate job.
+ 2. It would be difficult for implementations that include mitigation to share resources between tasks. For example, if different tasks represent different trotter step counts, there would need to be a complicated mechanism to share learning resources---that are specific to the application circuits---between multiple jobs.
+
```

Comment:
Another possible migration path is to introduce a new version, e.g. `EstimatorV1`. Importing an `Estimator` will default to V1, but people can choose to import the old version.